### ISSUE TITLE
Use ruff instead of flake8 + isort

### ISSUE THREAD

**Issue by niksirbi** [SainsburyWellcomeCentre/WAZP#30]
In the newest version of the [python-cookiecutter](https://github.com/SainsburyWellcomeCentre/python-cookiecutter) we switched to using [`ruff`](https://github.com/charliermarsh/ruff) for linting.

This has several advantages:
* runs much faster
* reduces the necessary config files (can be configured in `pyproject.toml`)
* replaces the functionality of both `isort` and `flake8`, so reduces the number of tools/dependencies
* it shouldn't alter our workflow (in terms of pre-commit hooks and actions).

If you agree @samcunliffe and @sfmig , I can implement the necessary changes.

**Comment by samcunliffe** [comment]
I mean, it's fine by me but I'm not a daily dev here so my opinion should be worth less.
Maybe open a PR and we can have a play?

FWIW `ruff` is [an 🟠 tool here](https://github.com/UCL-ARC/python-tooling/blob/main/pages/linting.md), and `isort` is 🟢 . Doesn't mean much more than other people in ARC would seem to prefer `isort`.

**Comment by samcunliffe** [comment]
Trivial extra comment, but I'd reword [this](https://hackmd.io/SbO3WyePSF-XEJVLZoX3og#Formatting-and-QC)...

> We use a common suite of linters to ensure a consistent code style. For python, our preferred list is [here](https://github.com/SainsburyWellcomeCentre/python-cookiecutter/blob/main/.pre-commit-config.yaml).

... assuming these guidelines are to be used for other projects and need to be up to date.

**Comment by sfmig** [comment]
no preference for me. I have not used `ruff` before but happy to give it a go if you think it will be better

**Comment by niksirbi** [comment]
> FWIW `ruff` is [an orange_circle tool here](https://github.com/UCL-ARC/python-tooling/blob/main/pages/linting.md), and `isort` is  . Doesn't mean much more than other people in ARC would seem to prefer `isort`.

I didn't know about this list, it's cool that it exists. I saw `flake8` is marked 🔴 there.
But there is no need to hurry, we can wait and see if we encounter any problems with ruff. I'd suggest delivering v0 with the current set of tools, and consider switching to `ruff` for v1. Will leave this issue open as a reminder to do that.

**Comment by niksirbi** [comment]
> Trivial extra comment, but I'd reword [this](https://hackmd.io/SbO3WyePSF-XEJVLZoX3og#Formatting-and-QC)...

Good point, I'll reword it
